barretenberg/cpp/src/barretenberg/ultra_honk/decider_prover.cpp

```diff
@@ -33,7 +33,7 @@ template <IsUltraFlavor Flavor> void DeciderProver_<Flavor>::execute_relation_ch
 
         PROFILE_THIS_NAME("sumcheck.prove");
         if constexpr (Flavor::HasZK) {
-            auto commitment_key = std::make_shared<CommitmentKey>(proving_key->proving_key.circuit_size);
+            auto commitment_key = std::make_shared<CommitmentKey>(Flavor::BATCHED_RELATION_PARTIAL_LENGTH);
```

Comment:
in a follow-up PR you would be able to refine this based on whether a realocation is needed